lib/Basic.php

```diff
@@ -90,18 +90,22 @@ public function status()
      */
     private function _init()
     {
-        global $conf, $page_output, $session;
+        global $conf, $page_output, $session, $notification;
         $passedToken = $this->_vars->get('token', '');
-        $sessionToken = $GLOBALS['session']->getToken();
 
         // Get the backend details.
         $backend_key = $this->_vars->backend;
         if (!isset($this->_backends[$backend_key])) {
             $backend_key = null;
         }
 
-        if ($backend_key && $this->_vars->submit && $session->checkToken($passedToken)) {
-            $this->_changePassword($backend_key);
+        if ($backend_key && $this->_vars->submit) {
+            try {
```

Comment:
I don't yet see why the `try/catch` block is needed, but it is probably better this way anyway.

Btw. you don't catch exceptions from the `$session->checkToken` call, right? Because that throws `Horde_Exception`.
Maybe then it should not even be in the `try/catch` block?

Reply:
Modern PHP can have multiple catch blocks (if you handle Horde_Auth_Exception with a more specific message/reaction than Horde_Exception) or union exception types like Horde_Auth_Exception|Horde_Exception - which might seem not needed as Horde_Auth_Exception is a Horde_Exception but communicates intend to the programmer reading the code.

Reply:
Thanks for catching that. It's a copy-paste mistake, it should be `Horde_Exception`.